### PR TITLE
Closes Issues #237 and #248

### PR DESCRIPTION
**Pull request by MichaelPantoja**
Chat code has been moved to it's own ChatController.java file and the name and host fields in chat have been changed for clarity. Additionally, a boolean value has been added for the chat connection success or failure. If the chat connection is unsuccessful, an alert will be given stating there was unsuccessful connection. The logic for the chat connection was not implemented.

### CODE REVIEW

File: src/edu/wright/cs/raiderplanner/model/Extension.java [hunk 1]
```diff
@@ -45,39 +45,71 @@ protected void replace(VersionControlEntity receivedVce) {
 		super.replace(receivedVce);
 	}
 
+	/**
+	 * Enumerated values for approval statuses.
+	 * @author unknown
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/Extension.java]: The Git history will allow you to see who authored this code. Please have a look and update this comment.

Comment (MichaelPantoja) [reply src/edu/wright/cs/raiderplanner/model/Extension.java]: How can I find it in the git history? I've undone the accidental changes to the metadata, just need to do this now.

Comment (rsanchez-wsu) [reply src/edu/wright/cs/raiderplanner/model/Extension.java]: In the CLI, you can do `git blame path/to/file` and then look for the specific lines. In Eclipse, right click on the line number and select `Show Revision Info`. Then you hover over the line number to see who introduced the line and when.

File: src/edu/wright/cs/raiderplanner/controller/MenuController.java [hunk 2]
```diff
@@ -1266,16 +1236,14 @@ protected void updateItem(final Task item, final boolean empty) {
...
-
 		Button check = new Button("Toggle complete");
 		check.getStyleClass().add("set-button");
 		check.setDisable(true);
 
 		Button delete = new Button("Remove");
 		delete.setDisable(true);
-
+		Button addNew = null;
+    addNew = new Button("Add a new task");
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/controller/MenuController.java]: Please fix indentation.

File: src/edu/wright/cs/raiderplanner/model/Building.java [hunk 3]
```diff
@@ -47,67 +47,78 @@ protected void replace(VersionControlEntity receivedVce) {
...
+	 * @return the name of the building
 	 */
 	public String getName() {
 		return name;
 	}
 
 	/**
-	 * Getter for Code.
-	 */
+     * Returns the code of the building as a String.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/Building.java]: Fix indentation.

File: src/edu/wright/cs/raiderplanner/model/Building.java [hunk 4]
```diff
@@ -47,67 +47,78 @@ protected void replace(VersionControlEntity receivedVce) {
...
+     * @return the code of the building
+     */
 	public String getCode() {
 		return code;
 	}
 
 	/**
-	 * Getter for latitude.
-	 */
+     * Returns the latitude of the building as a double.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/Building.java]: Fix indentation

File: src/edu/wright/cs/raiderplanner/model/Building.java [hunk 5]
```diff
@@ -47,67 +47,78 @@ protected void replace(VersionControlEntity receivedVce) {
...
+     * @return the latitude of the building
+     */
 	public double getLatitude() {
 		return latitude;
 	}
 
 	/**
-	 * Getter for longitude.
-	 */
+     * Returns the longitude of the building as a double.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/Building.java]: Fix indentation

File: src/edu/wright/cs/raiderplanner/model/Building.java [hunk 6]
```diff
@@ -47,67 +47,78 @@ protected void replace(VersionControlEntity receivedVce) {
...
+	 * @param newName the new name of the building
 	 */
 	public void setName(String newName) {
 		name = newName;
 	}
 
 	/**
-	 * Setter for Code.
-	 */
+     * Sets the code of the building.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/Building.java]: Fix indentation.

File: src/edu/wright/cs/raiderplanner/model/Building.java [hunk 7]
```diff
@@ -47,67 +47,78 @@ protected void replace(VersionControlEntity receivedVce) {
...
+     * @param newCode the new code of the building
+     */
 	public void setCode(String newCode) {
 		code = newCode;
 	}
 
 	/**
-	 * Setter for latitude.
-	 */
+     * Sets the latitude of the building.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/Building.java]: Fix indentation

File: src/edu/wright/cs/raiderplanner/model/Building.java [hunk 8]
```diff
@@ -47,67 +47,78 @@ protected void replace(VersionControlEntity receivedVce) {
...
+     * @param newLatitude the new latitude of the building
+     */
 	public void setLatitude(double newLatitude) {
 		latitude = newLatitude;
 	}
 
 	/**
-	 * Setter for longitude.
-	 */
+     * Sets the longitude of the building.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/Building.java]: Fix indentation.

File: src/edu/wright/cs/raiderplanner/model/StudyProfile.java [hunk 9]
```diff
@@ -52,19 +55,19 @@
 	}
 
 	/**
-	 * This method gets a list of Milestones.
-	 * @return returns an array of type Milestones
-	 */
+     * Returns an array of the Milestones of this StudyProfile.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/StudyProfile.java]: Please fix indentation.

File: src/edu/wright/cs/raiderplanner/model/StudyProfile.java [hunk 10]
```diff
@@ -52,19 +55,19 @@
...
 		Milestone[] milestone = new Milestone[this.milestones.size()];
 		milestone = this.milestones.toArray(milestone);
 		return milestone;
 	}
 
 	/**
-	 * This method gets a list of  Application Extension.
-	 * @return returns an array of type Application Extension
-	 */
+     * Returns an array of the extensions of this StudyProfile.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/StudyProfile.java]: Please fix indentation.

File: src/edu/wright/cs/raiderplanner/model/StudyProfile.java [hunk 11]
```diff
@@ -81,10 +84,10 @@
 	}
 
 	/**
-	 * This method gets a ArrayList of Tasks.
-	 * These Task themselves can contain a list of their own Task.
-	 * @return  an ArrayList of tasks
-	 */
+     * Returns an ArrayList of the Tasks of this StudyProfile.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/StudyProfile.java]: Please fix indentation.

File: src/edu/wright/cs/raiderplanner/model/StudyProfile.java [hunk 12]
```diff
@@ -121,24 +124,25 @@ public void addEventToCalendar(Event event) {
 	}
 
 	/**
-	 * Get Name.
-	 */
+     * Returns the name of this StudyProfile as a String.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/StudyProfile.java]: Please fix indentation.

File: src/edu/wright/cs/raiderplanner/model/StudyProfile.java [hunk 13]
```diff
@@ -121,24 +124,25 @@ public void addEventToCalendar(Event event) {
...
+     */
 	public String getName() {
 		return name;
 	}
 
 	/**
-	 * Get Year.
-	 * @return
-	 */
+     * Returns the year of this StudyProfile as an int.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/StudyProfile.java]: Please fix indentation.

File: src/edu/wright/cs/raiderplanner/model/StudyProfile.java [hunk 14]
```diff
@@ -121,24 +124,25 @@ public void addEventToCalendar(Event event) {
...
+     */
 	public int getYear() {
 		return year;
 	}
 
 	/**
-	 * Get Semester Number.
-	 * @return
-	 */
+     * Returns the semester number of this StudyProfile as an int.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/StudyProfile.java]: Please fix indentation.

**Pull request merged**